### PR TITLE
Build everything internally & fix apache builds

### PR DESCRIPTION
**Pull request by notcarbide**
Title explains it all. :)

(This was sitting locally for some time now without a commit, so I'm just kinda submitting as-is, but I can also break up the patches if needed.)

### CODE REVIEW

**Comment by NicolasCARPi** [comment]
Thanks I'll have a more thorough look later. As a quick thing that I spotted, use `$(pwd)` instead of ``pwd`` as it is POSIX. See https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts

**Comment by notcarbide** [comment]
Excuse the mess haha. Most of what I know about bash (and coding in general) was interpreted from XDA-Developers, so it's all very likely to be all over the place. POSIX probably isn't the only good practice I'm not following ;)

File: build.sh [hunk 1]
```diff
@@ -3,26 +3,77 @@
...
+    echo ""
 fi
 
-zip -r "$ZIP_PATH" assets index.php plugins src/classes themes vendor version
+# check for successful zip creation
+if [ -f "$ZIP_FILE" ]; then
+    echo "Build in $BUILD_DIR successful!"
+    echo "Output: $ZIP_FILE"
+else
+    echo "Build command failed!"
```
Comment (NicolasCARPi) [inline build.sh:73]: You should add an "exit 1" here

Comment (NicolasCARPi) [inline build.sh:73]: Instead of having REAL_OPTION, wouldn't it be better to check if ZIP_APPEND is empty?

File: build.sh [hunk 2]
```diff
@@ -3,26 +3,77 @@
...
+    rm "$ZIP_FILE"
+fi
+# also attempt to make zip path if does not exist
+if [ ! -d "$ZIP_PATH" ]; then
+    mkdir -p "$ZIP_PATH"
+fi
+
+# check if command for apache was used
+# and only include files for relevant versions
+if [ -n "$1" ]; then
```
Comment (NicolasCARPi) [inline build.sh]: this isn't necessary imho

File: build.sh [hunk 3]
```diff
@@ -3,26 +3,77 @@
...
 ./node_modules/.bin/grunt
 # install php dependencies
 composer install
 
+# remove the previous zip (or it will just append to it)
+if [ -f "$ZIP_FILE" ]; then
+    rm "$ZIP_FILE"
+fi
+# also attempt to make zip path if does not exist
+if [ ! -d "$ZIP_PATH" ]; then
```
Comment (NicolasCARPi) [inline build.sh]: this should be moved after ZIP_PATH declaration

**Review by NicolasCARPi — CHANGES_REQUESTED** [review]

**Review by NicolasCARPi — APPROVED** [review]

**Comment by NicolasCARPi** [comment]
Thank you!

**Pull request merged**